### PR TITLE
DataGrid / TreeList - Return promise from addRow method

### PR DESCRIPTION
**Pull request by igormalzew**

### CODE REVIEW

File: testing/tests/DevExpress.ui.widgets.treeList/editing.tests.js [hunk 1]
```diff
@@ -286,6 +286,63 @@ QUnit.test('AddRow method should expand row and add item after parent', function
...
+
+    // assert
+    assert.equal(this.getVisibleRows().length, 1, '1 visible row');
+
+    // act
+    const addRowAfterDeferredResolve = (parentIds, index) => {
+        const parentId = parentIds[index];
+
+        if(parentId !== undefined) {
+            return this.addRow(parentId).done(addRowAfterDeferredResolve.bind(null, parentIds, index + 1));
```
Comment (vconst) [inline testing/tests/DevExpress.ui.widgets.treeList/editing.tests.js]: Why is the bind method used here?

**Review by vconst — CHANGES_REQUESTED** [review]
Add test with using promise in onInitNewRow

File: testing/tests/DevExpress.ui.widgets.treeList/editing.tests.js (hunk 1 above)
Comment (igormalzew) [reply testing/tests/DevExpress.ui.widgets.treeList/editing.tests.js]: Replaced with arrow function for more readability

**Comment by igormalzew** [comment]
@vconst Added test with using promise in onInitNewRow

**Review by vconst — APPROVED** [review]

**Review by Alyar666 — APPROVED** [review]

**Pull request merged**